src/irc/core/irc-servers.c

```diff
@@ -78,17 +78,23 @@ static int ischannel_func(SERVER_REC *server, const char *data)
 	IRC_SERVER_REC *irc_server = (IRC_SERVER_REC *) server;
 	char *chantypes, *statusmsg;
 
+	g_return_val_if_fail(data != NULL, FALSE);
+
+	/* empty string is no channel */
```

Comment:
This was just a common-case optimisation in my suggestion, I don't think you need it